### PR TITLE
Feature/admin panel new year

### PR DESCRIPTION
**Pull request by jimylja**

### CODE REVIEW

File: src/app/admin-panel/new-year/autotitle.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'autotitle'
+})
+export class TitlePipe implements PipeTransform {
+
+  transform(value: any, curTitle: string, input: any): any {
```
Comment (dmitryy90) [inline src/app/admin-panel/new-year/autotitle.pipe.ts]: where do you use input param?

File: src/app/admin-panel/new-year/new-year.component.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            }
+          );
+        }
+      }
+    );
+    if (this.transititionForm.status === 'VALID') {
+      this.newYearTransitition.transitClasses(formData);
+    }
+  }
+}
```
Comment (dmitryy90) [inline src/app/admin-panel/new-year/new-year.component.ts]: don't remove the last empty line

File: src/app/services/new-year.service.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+   * @param   formData object that contain new titles for classes
+   */
+  public transitClasses(formData) {
+    const request = this.getTransitRequest(formData);
+    this.createClasses(request.transitClassesQuery).subscribe(
+      res => {res.data
+       .forEach(
+         (newClass, index) => {request.bindPupilsQuery[index].newClassID = newClass.id; }
+        );
+              this.bindPupils(request.bindPupilsQuery).subscribe();
```
Comment (dmitryy90) [inline src/app/services/new-year.service.ts]: indentation

File: src/app/services/new-year.service.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      }
+     );
```
Comment (dmitryy90) [inline src/app/services/new-year.service.ts]: indentation

File: src/app/admin-panel/new-year/new-year.component.ts (hunk 2 above)
Comment (jimylja) [reply src/app/admin-panel/new-year/new-year.component.ts]: fixed

File: src/app/services/new-year.service.ts (hunk 3 above)
Comment (jimylja) [reply src/app/services/new-year.service.ts]: fixed

File: src/app/services/new-year.service.ts (hunk 4 above)
Comment (jimylja) [reply src/app/services/new-year.service.ts]: fixed

**Pull request merged**